### ISSUE TITLE
Out-of-System Fees (Audit Request)

### ISSUE THREAD

**Issue by atdservicebot** [cityofaustin/atd-data-tech#1951]
## Out-of-System Fees
_AKA # 49 Out of Process TCP Fees, # 211 New GF Subtype, ROW's #1 Priority, Outstanding Audit Request_

**Note:** We discovered that both these issues are part of the same user story.
    - #211 will be closed 

**Stakeholders:** Paloma, Kim, Mooney

**User Story:** As the ATD Cashier at Toomey, I need to record all fees i accept in one system of record. The few remaining fees not in AMANDA should be added ASAP. 

**User Story:** As the person paying an 'out of process' fee, I need to be able to pay those fees online.

**Future State:**
_When ROW staff need to collect an 'out of process' fee_
1. In AMANDA, create a new GF folder, choose subtype ATD
2. Add the project location (site address) to the folder property tab
3. Add people record types Applicant, and Billed-To to the people tab
4. **Need SME Input** Complete required info fields on the info tab
    - Kim was unsure if TCP Review staff will want any info tab fields. Perhaps the TCP #? 
5. Insert & bill the fee(s), provide customer with the invoice
    - The folder status will be 'Pending Payment'
6. The customer pays the fee
7. The folder status updates to 'Payment Complete' -End Process-

- Brandi is best to ask for a list of fees collected outside of AMANDA
- The General Fees (GF) folder in AMANDA is what other departments are using to record fees outside of a specific case folder
- TCP Review Fee is being collected manually
    - **Need SME Input**
    - Amount? Flat Fee - no logic
    - Name of Fee?
    - ATD Surcharge?
---------

Old Description: ROW Fee audit recommendation that out of process TCP review fees be captured in AMANDA ASAP. 
Two Options TBD:
1. Create new fee and apply it on existing folder types
2. Create new (very simple) folder and fee

Request Date: 2018-12-03 13:37:00

Request ID: DTS18-103160

Status: Backlog

Customer Priority: 0

Level of Effort: Minor

DTS URL: https://atd.knack.com/dts#service-requests/view-issue-details/5c058609a3cc99085ea2bb6f

*Migrated from [atd-amanda #49](https://github.com/cityofaustin/atd-amanda/issues/49)*

**Comment by atdservicebot** [comment]
*From atdservicebot on 2019-05-10*:

A backlog item has been created for this request in the service request portal and the ROW Backlog Trello board. 

Created By: Tracy Linder
Created Date: 2018-12-03 13:40:00

**Comment by atdservicebot** [comment]
*From atdservicebot on 2019-05-10*:

2/8/19 Backlog Grooming Meeting - Paloma suggested looking into the General Fee folder in AMANDA to see if this i a possible solution. 
Ryan Mooney said that on 2/21 we are meeting with DSD to discuss adding these fees to the SP folder. 

Created By: Tracy Linder
Created Date: 2019-02-08 10:27:00

**Comment by SurbhiBakshi** [comment]
- Question for Paloma - recommended best practice is to enter number of hours, and that will set off calculation with fee amount and 4%surcharge. Will that work for you? #1951. 
A - yes, this will work for ROW.
- Question - LIT said we could have them work on putting the Out of System fee on Portal in parallel with the creation of the fee in Amanda. Would that work for you?
A - yes, that will work, but main focus should be on getting the TCP fee into Amanda. Ryan M would like to see an option where an applicant did not need an account or to be associated with the folder to be able to pay this fee. 
- Question - need to confirm GL account numbers for the fees, who would be the right person to contact for that.
A - ask Van P, Paloma will confirm TCP GL number. Confirmed with Paloma

**Comment by SurbhiBakshi** [comment]
Follow up with CTM on which subtype code is appropriate

**Comment by Nadin-Nader** [comment]
Followed up with George on 4/13 and spoke about the concerns brought up by Raj in the meeting.
Concern for me to follow up on:
If a Fee is put into the GF folder for the SP folder will it show up on the SP folder or associate it with the SP folder.

This would be a big project as each folder has different processes that it follows and they would have to be assessed individually to change the process properly. Also this can cause issues if the GF folder has something unpaid and the process or user closes the parent or related folder.


If a Misc TCP fee has a tracking number of some sort, is there a field or a way to have that tracking number in AMANDA or can/does AMANDA track it.

This is possible but on the GL folder. The logic would have to be added  to the GF folder if a field is available

I also had him look at the demo for the TCP fee and asked for his feed back when he has a chance.

@SurbhiBakshi  George would like to be a part of the next meeting if possible.

**Comment by SurbhiBakshi** [comment]
[LIT work video ](https://drive.google.com/file/d/1z6MSiJgoxDmwHYxFxYJPBEdNR_9RhuQt/view?usp=sharing)

**Comment by amenity** [comment]
@raajeshb - can you please add an estimate for this?

**Comment by TracyLinder** [comment]
Surbhi thinks this is close to being completed. Work still needs to be moved to TEST, pending Sree's permissions issue being resolved.

**Comment by TracyLinder** [comment]
Per sree, this is in Dev right now, it needs to be testing in Dev before going to TEST. @Nadin-Nader can you setup some time with me to make sure I know my way around Dev?

**Comment by Nadin-Nader** [comment]
Scheduled meeting for UAT on DEV (using my AMANDA Dev with Kim driving if need be) for 5/18th/2020 @930am

**Comment by Nadin-Nader** [comment]
@TracyLinder  This is good to go to TEST and PROD. Had the meeting to UAT the out of system fees with Paloma and Kim. We had to use my Dev application since the installer did not work on their PC'S (need to have admin run batch file). We were able to share my Dev environment but logged in with Kim's credentials.

She worked through the folder to test the UAT and found that the fees are there and it is good to go.
**However, Paloma did notice that the Traffic Reviewer Info needed to be cleaned up by Just using the generic option of "Traffic Control Review" instead of having the individual names.
She also noticed that when the Fees are modified, deleted or recalculated for any reason (hours change or fee change), When you try to "Insert and Run", it doesn't calculate or insert the fees. They have to be inserted manually.**
-Paloma said that this particular little pain was not a must have or need but if we can get it done really quick and soon with no effort, then she would like the Insert and Run Function to work.**

I have attached a recording of the meeting. The second half of the recording is for #1518 
[https://web.microsoftstream.com/video/c540fe12-2fa6-4f21-b56f-6aad97c00079](url)

**Comment by TracyLinder** [comment]
@Nadin-Nader what ended up being the solution for these user stories? Will they be using the GF folder? Also is there a list of the fees that are being added that we can add to this issue for documentation purposes? Thank you!

**Comment by Nadin-Nader** [comment]
@TracyLinder the user story was; "As the person paying an 'out of process' fee, I need to be able to pay those fees online." or "As the ATD Cashier at Toomey, I need to record all fees i accept in one system of record. The few remaining fees not in AMANDA should be added ASAP."

These have now been resolved by having the correct fees showing online for the applicant to pay. Paloma was able to create, delete and reinsert (manually) all the necessary fees on the GF folder. This was of course all done after she created an application and paid for all of the fees online.
As far as the list of fees, i did not get one from them nor did they mention any other fees that needed to be added.  If there were additional fees added while fixing this, Sree or LIT would know which ones were added. I think that part may have been done before i got to it.
Let me know if you need anything else.

**Comment by TracyLinder** [comment]
@Nadin-Nader can you please add an estimate to this one? (Out of system fees)

**Comment by TracyLinder** [comment]
Also @Nadin-Nader, is the pipeline correct on this one?

**Comment by TracyLinder** [comment]
Currently moving to test and prod, then will be complete.

**Comment by TracyLinder** [comment]
Per Sree: Ready to move to production. Sree has sent an email w/ status and a demo.

**Comment by Nadin-Nader** [comment]
@TracyLinder @raajeshb 
I reviewed this and all seems well and ready to go to PROD but is there a way that we can address what Paloma mentioned "that when the Fees are modified, deleted or recalculated for any reason (hours change or fee change), When you try to "Insert and Run", it doesn't calculate or insert the fees. They have to be inserted manually."
-Paloma said that this particular little pain was not a must have or need but if we can get it done really quick and soon with no effort, then she would like the Insert and Run Function to work.

Please let me know so i can let Paloma know. I will also reach out to Sree.

**Comment by Nadin-Nader** [comment]
email to Sree:
From: Nader, Nadin 
Sent: Monday, June 8, 2020 4:08 PM
To: Sreelatha.b@launchitcorp.com; Linder, Tracy <Tracy.Linder@austintexas.gov>
Cc: Rajesh B. <Rajesh.b@launchitcorp.com>
Subject: RE: Trello Card # TCP Out of Process - Online payment - DEMO

Sree,

Hope you are doing well. I wanted to let you know that I have reviewed the demo and updated the card with my notes. It looked good to go to Prod but in my previous meeting with Paloma she had a quick question/request:

“when the Fees are modified, deleted or recalculated for any reason (hours change or fee change), When you try to "Insert and Run", it doesn't calculate or insert the fees. They have to be inserted manually."
-Paloma said that this particular little pain was not a must have or need but if we can get it done really quick and soon with no effort, then she would like the Insert and Run Function to work.

Please let me know so i can let Paloma know. 

Respectfully,

Nadin A. Nader 
IT Business Systems Analyst
Data and Technology Services 
Austin Transportation Department

**Comment by TracyLinder** [comment]
@Nadin-Nader I'm interested in Sree's response on this. If we need to open another ticket for Paloma's unresolved comment about the insert + run, that would probably be best, since it seems to be out of scope for this ticket.

**Comment by Nadin-Nader** [comment]
@TracyLinder 
Spoke to Sree via chat and she suggested to do that part later on. I sent an email and CC'd you but will add the Teams chat to these notes:

**TEAMS Chat:**

> [4:15 PM] Blaze Kannan, Sreelatha - Vendor
>     Hi Nadin, Can we have a meeting regarding 1951? 
> [4:17 PM] Nader, Nadin
>     I can chat about it real quick. did you have questions?
> ​[4:18 PM] Blaze Kannan, Sreelatha - Vendor
>     i dont have any questions as such, but i just want to make sure why Paloma using INSERT & RUN when hours info is aailable
> ​[4:18 PM] Blaze Kannan, Sreelatha - Vendor
>     Because it is never added to INSERT & RUN
> [4:19 PM] Blaze Kannan, Sreelatha - Vendor
>     while creating the folder itself fees gets added....later if they want to change the hours...they need to update hours...so that fee will be recalculated
> ​[4:19 PM] Nader, Nadin
>     what if there is a fee change?
> ​[4:20 PM] Nader, Nadin
>     will it still calculate it manually?
> ​[4:20 PM] Blaze Kannan, Sreelatha - Vendor
>     hours need to be updated. so that fees will be calculated automatically
> ​[4:21 PM] Blaze Kannan, Sreelatha - Vendor
>     say - 2 hours initially - 100 (per hour 50) 
> ​[4:21 PM] Blaze Kannan, Sreelatha - Vendor
>     later they thought its 4 hours...then 
> ​[4:21 PM] Blaze Kannan, Sreelatha - Vendor
>     info field must be updated as 4
> ​[4:21 PM] Blaze Kannan, Sreelatha - Vendor
>     so that fee will be re-calculated as 200 
> ​[4:22 PM] Blaze Kannan, Sreelatha - Vendor
>     hour flat fee changes, then lookup table needs to be updated by you/ctm with new flat fee
> ​[4:22 PM] Blaze Kannan, Sreelatha - Vendor
>     thats all
> [4:22 PM] Blaze Kannan, Sreelatha - Vendor
>     one time change
> ​[4:23 PM] Nader, Nadin
>     ok cool. then of ot recalculates it just by updating the hours, i think it should be good to go. WHat is the level of effort to put it in the Insert and Run function?
> ​[4:23 PM] Nader, Nadin
>     *then if it recalculates
> [4:25 PM] Blaze Kannan, Sreelatha - Vendor
>     it is not a huge effort...all that i need to do is, create a proc and use hours info and flat fee lookuptable ...however i dont suggest to do it now...because then i need to revert this commit...take all changes back to local host and do everything again and DEV and TEST....it is cumbersome process... though the effort is small
> ​[4:26 PM] Blaze Kannan, Sreelatha - Vendor
>     instead i suggest - lets move this PROD. create a new ticket and will take care of it in future
> ​[4:26 PM] Blaze Kannan, Sreelatha - Vendor
>     this makes everyones lives easier. 
> ​[4:27 PM] Blaze Kannan, Sreelatha - Vendor
>     if we identified this in our demo, then i would have made changes there and moved to DEV and to TEST....Now already the changes are in TEST. if I need to touch it.. I NEED TO REVERT BACK from TEST and DEV to LOCALHOST
> ​[4:27 PM] Blaze Kannan, Sreelatha - Vendor
>     hope you got it
> ​[4:31 PM] Nader, Nadin
>     Yes i totally understood and this wasnt a had to have. I think its a good idea for later once this gets implemented. I just want to inform Paloma and give her the  level of effort for that so she can have an idea for later.
> [4:32 PM] Blaze Kannan, Sreelatha - Vendor
>     sure 2 story points - again i do not suggest as it may not add much value. Going to INSERT & RUN and selecting the fee type is difficult than just giving the hours and save. (smile) 
> ​[4:32 PM] Blaze Kannan, Sreelatha - Vendor
>     Again users wish (smile)
> ​[4:33 PM] Blaze Kannan, Sreelatha - Vendor
>     because again, it is going to use the same hours mentioned in the info
> ​[4:51 PM] Nader, Nadin
>     Ok i will let her know. Thank you fo responding so fast (smile)

**Email sent**

> Hi Paloma/Kim,
> 
> Hope you are both doing well. I wanted to let you know that I spoke to Sree about my notes from the review of the TCP Out of System fees where you stated that “when the Fees are modified, deleted or recalculated for any reason (hours change or fee change), When you try to "Insert and Run", it doesn't calculate or insert the fees. They have to be inserted manually.” 
> 
> Sree mentioned that it has never been added to the “Insert and Run” since in this case changing the hours will recalculate the fees automatically. It wouldn’t be too much effort to change but she suggests to add it later on once this goes to PROD and you like the function and how it flows. She would need to create a process and use the hours info and flat fee lookuptable. 
> 
> Also since the change is already in TEST, the changes would need to get reverted and she will need to start everything from DEV again. Once we get this to PROD you can submit a new request to have the new process added to the “Insert and Run”
> 
> Please let me know if you are good with this as we just need to give the go ahead for PROD since all went well in TEST.
> 
> 
> Respectfully,
> 
> Nadin A. Nader 
> IT Business Systems Analyst
> Data and Technology Services 
> Austin Transportation Department
>

**Comment by Nadin-Nader** [comment]
@TracyLinder @raajeshb  Response from Paloma:

> From: Amayo-Ryan, Paloma <Paloma.Amayo-Ryan@austintexas.gov> 
> Sent: Monday, June 8, 2020 7:15 PM
> To: Nader, Nadin <Nadin.Nader@austintexas.gov>; Perez, Kimberly <Kimberly.Perez@austintexas.gov>
> Cc: Linder, Tracy <Tracy.Linder@austintexas.gov>
> Subject: RE: (1951) TCP Out of Process fees UAT on Dev and (1518) Unpaid Flag Fee on Permit Issuance
> 
> Hey Nate!
> 
> I’m on board with holding off as Sree suggests.  Thanks for checking in on this!

**Sent Sree an email informing her to deploy to PROD**

READY FOR DEPLOYMENT TO PROD

**Comment by Nadin-Nader** [comment]
Deployment has been split into two requests:

One is currently with the AMANDA CM Team in their "To-Do Pipeline" (SCTASK0073676). Waiting for the deployment and confirmation to PROD
The manual deployment was to add a lookup to PROD from DEV and TEST (51060 Lookup2=48). I was able to have Andrea from Team Tango add the lookup. SCTASK0074245 she will close this task out as it has been completed.
Copying these notes to #2406 as well

Waiting on CM Team to deploy

**Comment by Nadin-Nader** [comment]
Paloma will have users test and get back to us next week to give an update.

**Comment by Sreelatha-BlazeKannan** [comment]
It is in PROD Now.

**Comment by TracyLinder** [comment]
**Email to Stakeholders:**

Good Afternoon, 

Some good news today. The work for #1951 for the GF folder for out of system fees is now in Production! These fees are also payable on the portal. Yay!

Please let us know if there are any questions. 

All the Best,

Tracy Linder, Product Manager

**Comment by Nadin-Nader** [comment]
Hi Paloma/Kim,

Hope you are both doing well and staying safe. I was reaching out to you regarding the Out of System Fees #2406 and #1951 to see if you were able to gather enough information from your users UAT’ing the Out of system fees changes. Please let me know the outcome as soon as you can.

Respectfully,

Nadin A. Nader 
IT Business Systems Analyst
Data and Technology Services 
Austin Transportation Department

**Comment by TracyLinder** [comment]
@Nadin-Nader the comments say this is already in PROD and UAT passed. Is your email to Paloma & Kim correct?

**Comment by TracyLinder** [comment]
Laura Roy is the primary user of this feature, she may not begin using it until next week (7/13)
